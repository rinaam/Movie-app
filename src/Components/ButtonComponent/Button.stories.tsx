import React from "react";
import { Story, Meta } from "@storybook/react";

import ButtonComponent, {
  IButtonProps,
} from "./ButtonComponent";
import RandomIcon from '../../Assets/Icons/RandomIcon.svg';
import CloseIcon from '../../Assets/Icons/CloseIcon.svg';

export default {
  title: "Button",
  component: ButtonComponent,
} as Meta;

const Template: Story<IButtonProps> = (args) => <ButtonComponent {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  variant: "primary",
  children: "Roll",
};

export const Round = Template.bind({});
Round.args = {
  variant: "round",
  children: "Load",
};

export const Floating = Template.bind({});
Floating.args = {
  variant: "floating",
  children: <RandomIcon />,
};

export const Close = Template.bind({});
Close.args = {
  variant: "close",
  children: <CloseIcon />,
};
