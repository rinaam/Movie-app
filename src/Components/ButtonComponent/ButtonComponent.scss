@use "sass:map";
@import "../../Styles/variables.scss";
@import "../../Styles/breakpoints.scss";

.btn {
  font-weight: map.get($font-weight, "s");
  border: 0;
  cursor: pointer;
  display: inline-block;
  line-height: 1;
  padding: map.get($spacing, "s");
  outline: none;
  &-primary {
    color: $white;
    height: 35px;
    background-color: $black;
  }
  &-round {
    width: 60px;
    height: 60px;
    border-radius: 100%;
    background-color: $white-smoke;
    margin: map.get($spacing, "s");
  }
  &-floating {
    @extend .btn-round;
    position: fixed;
    right: 0;
    bottom: 0;
    margin: map.get($spacing, "m");
    color: $white;
    background-color: $black;
    font-size: map.get($font-size, "m");
    box-shadow: map.get($box-shadow, "primary");
    @include breakpoint(smallDevices) {
      margin: map.get($spacing, "xl");
    }
  }
  &-close {
    background-color: $white;
  }
}