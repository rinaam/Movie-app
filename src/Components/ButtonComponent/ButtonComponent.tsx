import React from 'react';
import './ButtonComponent.scss';

export interface IButtonProps {
    variant: 'primary' | 'round' | 'floating' | 'close';
    children: JSX.Element | string;
    onClick: () => void;
}

const ButtonComponent: React.FC<IButtonProps> = ({
  variant,
  children,
}) => (
  <button
    type="button"
    className={`btn btn-${variant}`}
  >
    {children}
  </button>
);

export default ButtonComponent;
