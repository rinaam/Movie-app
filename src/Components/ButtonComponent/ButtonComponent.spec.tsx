import { act, create } from 'react-test-renderer';
import React from 'react';
import ButtonComponent from './ButtonComponent';

describe('ButtonComponent', () => {
  it('should render ButtonComponent', () => {
    const fakeClick = jest.fn();
    const wrapper = create(
      <ButtonComponent variant="primary" onClick={fakeClick}>Roll</ButtonComponent>,
    ).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it('should trigger click', () => {
    const fakeClick = jest.fn();
    const wrapper = create(
      <ButtonComponent variant="primary" onClick={fakeClick}>Roll</ButtonComponent>,
    );

    act(() => {
      wrapper.root.props.onClick();
    });

    expect(fakeClick).toBeCalled();
  });
});
