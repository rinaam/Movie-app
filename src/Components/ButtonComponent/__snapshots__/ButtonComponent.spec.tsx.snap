// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonComponent should render ButtonComponent 1`] = `
<button
  className="btn btn-primary"
  onClick={[MockFunction]}
  type="button"
>
  Roll
</button>
`;
